engine/apps/mobile_app/tasks.py

```diff
@@ -499,3 +500,192 @@ def conditionally_send_going_oncall_push_notifications_for_schedule(schedule_pk)
+    """
+
+    shift_swap_requests_in_notification_window = []
+    for shift_swap_request in ShiftSwapRequest.objects.filter(benefactor__isnull=True, swap_start__gt=now):
+        notification_window_start = max(
+            shift_swap_request.created_at, shift_swap_request.swap_start - SSR_EARLIEST_NOTIFICATION_OFFSET
+        )
+        notification_window_end = min(notification_window_start + SSR_NOTIFICATION_WINDOW, shift_swap_request.swap_end)
+
+        if notification_window_start <= now <= notification_window_end:
+            shift_swap_requests_in_notification_window.append(shift_swap_request)
+
+    return shift_swap_requests_in_notification_window
+
+
+@shared_dedicated_queue_retry_task(autoretry_for=(Exception,), retry_backoff=True, max_retries=MAX_RETRIES)
+def notify_shift_swap_request(shift_swap_request_pk: int) -> None:
+    """
+    Notify relevant users for an individual shift swap request.
+    """
+    try:
+        shift_swap_request = ShiftSwapRequest.objects.get(pk=shift_swap_request_pk)
+    except ShiftSwapRequest.DoesNotExist:
+        logger.info(f"ShiftSwapRequest {shift_swap_request_pk} does not exist")
+        return
+
+    now = timezone.now()
+    for user in shift_swap_request.possible_benefactors:
+        if _should_notify_user_about_shift_swap_request(shift_swap_request, user, now):
+            notify_user_about_shift_swap_request.delay(shift_swap_request.pk, user.pk)
+            _mark_shift_swap_request_notified_for_user(shift_swap_request, user)
+
+
+@shared_dedicated_queue_retry_task(autoretry_for=(Exception,), retry_backoff=True, max_retries=MAX_RETRIES)
+def notify_user_about_shift_swap_request(shift_swap_request_pk: int, user_pk: int) -> None:
+    """
+    Send a push notification about a shift swap request to an individual user.
+    """
+    # avoid circular import
+    from apps.mobile_app.models import FCMDevice, MobileAppUserSettings
+
+    try:
+        shift_swap_request = ShiftSwapRequest.objects.get(pk=shift_swap_request_pk)
+    except ShiftSwapRequest.DoesNotExist:
+        logger.info(f"ShiftSwapRequest {shift_swap_request_pk} does not exist")
+        return
+
+    try:
+        user = User.objects.get(pk=user_pk)
+    except User.DoesNotExist:
+        logger.info(f"User {user_pk} does not exist")
+        return
+
+    device_to_notify = FCMDevice.get_active_device_for_user(user)
+    if not device_to_notify:
+        logger.info(f"FCMDevice does not exist for user {user_pk}")
+        return
+
+    try:
+        mobile_app_user_settings = MobileAppUserSettings.objects.get(user=user)
+    except MobileAppUserSettings.DoesNotExist:
+        logger.info(f"MobileAppUserSettings does not exist for user {user_pk}")
+        return
+
+    if not mobile_app_user_settings.info_notifications_enabled:
+        logger.info(f"Info notifications are not enabled for user {user_pk}")
+        return
+
+    if shift_swap_request.status != ShiftSwapRequest.Statuses.OPEN:
```

Comment:
_nit_: we could create a `@property` on `ShiftSwapRequest` as such:
```python
@property
def is_open(self) -> bool:
  return not any([self.is_deleted, self.is_taken, self.is_past_due])
```

Reply:
Good call, added in https://github.com/grafana/oncall/pull/2717/commits/f747ec19c850a963e8db1223f5e72499c0d88d2a